CHANGES.md

```diff
@@ -1,5 +1,124 @@
 # Changes
 
+## stable-2.12.0
+
+This release introduces route-based policy to Linkerd, allowing users to define
+and enforce authorization policies based on HTTP routes in a fully zero-trust
+way. These policies are built on Linkerd's strong workload identities, secured
+by mutual TLS, and configured using types from Kubernetes's [Gateway API](https://gateway-api.sigs.k8s.io/).
+
+The 2.12 release also introduces access logging, a long-awaited feature that
+optionally allows Linkerd to produce Apache-style request logs, optional
+support for `iptables-nft`, and a host of other improvements and performance
+enhancements.
+
+**Upgrade notes**: Please see the [upgrade instructions](upgrade-2120). <!-- TODO fix link -->
+
+* Proxy
+  * Added a `config.linkerd.io/shutdown-grace-period` annotation to configure
+   the proxy's maximum grace period for graceful shutdown
+  * Added a new `iptables-nft` mode for `proxy-init`
+  * Fixed an issue where the proxy-injector would break when using
+    `nodeAffinity` values for the control plane
+
+* Control Plane
+  * Added support for per-route policy by supporting AuthorizationPolicy
+    resources which target HttpRoute resources
+  * Removed kube-system exclusions from watchers to fix service discovery
+    for workloads in the kube-system namespace (thanks @JacobHenner)
+  * Updated healthcheck to ignore `Terminated` state for pods (thanks
+    @AgrimPrasad!)
+  * Updated the default policy controller log level to `info`; the controller
+    will now emit INFO level logs for some of its dependencies
+  * Added probe authorization by default, allowing clusters that use a
+    default `deny` policy to not explicitly need to authorize probes
+  * Fixed an issue where certain control plane components were not restarting
+    as necessary after a trust root rotation
+  * Fixed an issuer where the `--default-inbound-policy` setting was not
+    being respected
+
+* CLI
+  * Fixed the `linkerd check` command crashing when unexpected pods are found
+    in a Linkerd namespace
+  * Updated the `linkerd authz` command to support AuthorizationPolicy and
+    HttpRoute resources
+  * Updated linkerd check to allow RSA signed trust anchors (thanks
+    @danibaeyens)
```

Comment:
```suggestion
  * Updated `linkerd check` to allow RSA signed trust anchors (thanks
    @danibaeyens)
```

nit: consistency. Perhaps we should decide on a standard here and enforce it throughout the section (i.e CLI commands should or shouldn't be in backticks). Wdyt?